packages/gestalt/src/PageHeader.js

```diff
@@ -0,0 +1,72 @@
+// @flow strict
+import React, { type Node, Children } from 'react';
+import Box from './Box.js';
+import Flex from './Flex.js';
+import Heading from './Heading.js';
+import Text from './Text.js';
+import { type Dimension } from './boxTypes.js';
+
+type Props = {|
+  title: string,
+  maxWidth?: Dimension,
+  primaryAction?: Node,
+  secondaryAction?: Node,
+  subtext?: string,
+|};
+
+export default function PageHeader({
+  maxWidth = '100%',
+  primaryAction,
+  secondaryAction,
+  subtext,
+  title,
+}: Props): Node {
+  const renderAction = (action?: Node): Node => {
+    // This is technically all temporary, until we create a proper Menu component
+    // Then, we could type check using Flow for Buttons, Links, or Menus
```

Comment:
That's true, though we can use Flow right now to check for `IconButton | Button | Link | Tooltip` and obviate all this logic

Reply:
But how do we do that and account for the React.Fragment needed when using Dropdowns?